### PR TITLE
scripts: kconfig: Turn most warnings into errors

### PR DESCRIPTION
**Pull request by ulfalizer**
In particular, this will turn assignments to undefined Kconfig variables
into errors, which are very easy to miss otherwise (e.g. when Kconfig
symbols get renamed or removed).

Warnings generated by anything tested by CI (scripts/sanitycheck) will
get caught.

Have a whitelist of warnings that are not turned into errors. Some
warnings currently whitelisted should be turned into errors as well, but
would require a bit of work.

### CODE REVIEW

**Comment by ulfalizer** [comment]
CC @carlescufi @SebastianBoe @nashif

File: scripts/kconfig/kconfig.py [hunk 1]
```diff
@@ -5,6 +5,35 @@
...
+    "unsatisfied direct dependencies",
+
+    # Assignments to certain undefined symbols. These are only defined for ARC
+    # and x86, respectively, but are set in some "shared" .conf files.
+    "undefined symbol ARC_INIT",  # Issue #7977
+    "undefined symbol SSE",       # Issue #7978
+
+    # Warning generated when a symbol is set more than once in a .config file.
+    # These should be investigated, but whitelist them for now.
+    "set more than once",
```
Comment (SebastianBoe) [inline scripts/kconfig/kconfig.py]: Merging two different Kconfig fragments with overlapping and agreeable requirements should not produce a warning.

Comment (ulfalizer) [reply scripts/kconfig/kconfig.py]: If someone messes up and puts in the same assignment twice, it's cool to point it out.

There are some cases where it makes sense to have more than one assignment to the same variable though, when merging different combinations of `.config` files. Might even be okay to override assignments from the base configuration with a different value in some cases.

It's a tradeoff. If you think those redundant assignment warnings (when you give the same value both times) are more annoying than helpful, I could disable them though.

**Comment by ulfalizer** [comment]
CC @jarz-nordic (3f99eef)

**Comment by ulfalizer** [comment]
@inakypg @nashif @andrewboie 
What's the purpose of the `CONFIG_NUM_IRQS=2` assignment in `lib/sprintf/prj.conf`? Is it needed?

It's causing a warning for ARC, where `NUM_IRQS` [has a `range`]( http://docs.zephyrproject.org/reference/kconfig/CONFIG_NUM_IRQS.html#kconfig-definitions):

    Using /home/buildslave/src/github.com/zephyrproject-rtos/zephyr/boards/arc/em_starterkit/em_starterkit_em7d_defconfig as base
    Merging /home/buildslave/src/github.com/zephyrproject-rtos/zephyr/tests/lib/sprintf/prj.conf
    ...
    Error: Aborting due to non-whitelisted Kconfig warning 'warning: user value 2 on the int symbol NUM_IRQS (defined at ...) ignored due to being outside the active range ([17, 256]) -- falling back on defaults'.

From https://app.shippable.com/github/zephyrproject-rtos/zephyr/runs/17015/1/console.

**Comment by andrewboie** [comment]
> What's the purpose of the CONFIG_NUM_IRQS=2 assignment in lib/sprintf/prj.conf? Is it needed?

Cruft. I sent #8347 .

**Review by tyrantlizard77 — APPROVED** [review]

**Comment by ulfalizer** [comment]
recheck

**Review by nashif — APPROVED** [review]

File: scripts/kconfig/kconfig.py (hunk 1 above)
Comment (SebastianBoe) [reply scripts/kconfig/kconfig.py]: I do. Please disable.

**Pull request merged**